test/helpers/patient_query_helper_test.rb

```diff
@@ -899,5 +899,125 @@ class PatientQueryHelperTest < ActionView::TestCase
     filtered_patients_array = [patient_1, patient_4]
     assert_equal filtered_patients_array.map { |p| p[:id] }, filtered_patients.pluck(:id)
   end
+
+  # --- PATIENTS TABLE DATA TESTS --- #
+
+  test 'patients table data filters out current monitoree' do
+    Patient.destroy_all
+    user = create(:public_health_enroller_user)
+
+    patient_1 = create(:patient, creator: user)
+    patient_2 = create(:patient, creator: user)
+    patient_3 = create(:patient, creator: user)
+
+    patients = Patient.all
+
+    exclude_patient_id = patient_1.id
+    params = ActionController::Parameters.new({
+      query: {
+        search: "",
+        entries: 5,
+        workflow: "global",
+        tab: "all",
+        scope: "all",
+        tz_offset: 240,
+        exclude_patient_id: exclude_patient_id,
+        filter: [{
+          dateOption: nil,
+          filterOption: {
+            description: "Monitorees that are a Head of Household or self-reporter",
+            name: "hoh",
+            title: "Daily Reporters (Boolean)",
+            type: "boolean"
+          },
+          value: true
+        }]
+      }
+    })
+
+    # Check for monitorees that are HoH or self-reporter
+    patients = patients_table_data(params, user)
+    patients_array = [ patient_2, patient_3 ]
+    assert_equal patients_array.map { |p| p[:id] }, patients[:linelist]&.pluck(:id)
+
+    # Check that current monitoree is not in patients list
+    patients_by_id = patients[:linelist]&.pluck(:id)
+    assert_not_includes(patients_by_id, exclude_patient_id)
+  end
+
+  test 'patients table data returns patients when exclude_patient_id is nil' do
+    Patient.destroy_all
+    user = create(:public_health_enroller_user)
+
+    patient_1 = create(:patient, creator: user)
+    patient_2 = create(:patient, creator: user)
+    patient_3 = create(:patient, creator: user)
+
+    patients = Patient.all
+
+    params = ActionController::Parameters.new({
+      query: {
+        search: "",
+        entries: 5,
+        workflow: "global",
+        tab: "all",
+        scope: "all",
+        tz_offset: 240,
+        filter: [{
+          dateOption: nil,
+          filterOption: {
+            description: "Monitorees that are a Head of Household or self-reporter",
+            name: "hoh",
+            title: "Daily Reporters (Boolean)",
+            type: "boolean"
+          },
+          value: true
+        }]
+      }
+    })
+
+    # Check for monitorees that are HoH or self-reporter
+    patients = patients_table_data(params, user)
+    patients_array = [ patient_1, patient_2, patient_3 ]
+    assert_equal patients_array.map { |p| p[:id] }, patients[:linelist]&.pluck(:id)
+  end
+
+  test 'patients table data returns patients when exclude_patient_id is not valid' do
```

Comment:
with validation, this should return a 400 instead

Reply:
changes to the test look good but don't forget to update the description as well!

Reply:
test changes look good but don't forget to update the description as well!